openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -284,11 +285,20 @@ def get(self, request):
     def list(self, request):
         """
         GET /api/user/v1/accounts?username={username1,username2}
+        GET /api/user/v1/accounts?email={user_email}
         """
         usernames = request.GET.get('username')
+        user_email = request.GET.get('email')
         try:
             if usernames:
                 usernames = usernames.strip(',').split(',')
+            elif user_email:
+                try:
+                    user_email = user_email.strip('')
+                    user = User.objects.get(email=user_email)
+                    usernames = [user.username]
+                except UserNotFound:
+                    return Response(status=status.HTTP_404_NOT_FOUND)
             account_settings = get_account_settings(
                 request, usernames, view=request.query_params.get('view'))
         except UserNotFound:
```

Comment:
This could use some cleanup. Two options
1) Pull the `usernames` creation out of the parent `try` block (probably should be have true before this PR). Then you keep your `try` block outside of the parent `try` block. 
2) You remove the error catching you have because it's the same as the parent one which will already catch it. (`except UserNotFound => 404`) 

I think 1 is probably cleaner because it doesn't depend on the fact that they happen to have the same exception.